### ISSUE TITLE
Incorrect warning on cast to non-nullable reference type

### ISSUE THREAD

**Issue by jcouv** [dotnet/roslyn#40111]
In the following code sample, I expect no warning on the cast.

``` csharp

#nullable enable
public class C 
{
    public void M(object? o) 
    {
        _ = (object)o == null; // warning CS8600: Converting null literal or possible null value to non-nullable type.
    }
}
```

**Comment by RikkiGibson** [comment]
At least recently I assumed this was by-design. Did we decide this is too onerous in practice? I would agree with that.

**Comment by jcouv** [comment]
@RikkiGibson You are right. I remembered the LDM decision wrong.
Here are notes confirming the current behavior: https://github.com/dotnet/roslyn/blob/master/docs/features/nullable-reference-types.md#explicit-cast

I'll go ahead and close.
